### ISSUE TITLE
SQLite3 returned ErrorConstraint while attempting to perform step: NOT NULL constraint failed: pool_owner_backup.pool_owner_index

### ISSUE THREAD

**Issue by stakeada** [cardano-foundation/cardano-wallet#1185]
# Context
 
Just run cardano-wallet as follows:

```
# cardano-wallet launch --genesis-block-hash $(cat genesis-hash.txt) -- --config itn_rewards_v1-config.yaml 
```

| Information  | -                                                     |
| ---          | ---                                                   |
| Version      | 2019.12.13 (git revision: 0000000000000000000000000000000000000000) |
| Platform     | Nixos          |
| Installation | Source          |

# Steps to Reproduce

Just run cardano-wallet and you'll get the following error in stdout - I've just built from **master** to solve #1184.

```
[cardano-wallet:Info:20] [2019-12-15 19:40:35.50 UTC] Jörmungandr is ready.     
SQLite3 returned ErrorConstraint while attempting to perform step: NOT NULL constraint failed: pool_owner_backup.pool_owner_index                               
cardano-wallet-jormungandr: SQLite3 returned ErrorConstraint while attempting to perform step: NOT NULL constraint failed: pool_owner_backup.pool_owner_index   


```

## Expected behavior

The cardano-node should not crash.

## Actual behavior

The cardano-node crashes.

**Comment by Fell-x27** [comment]
Cardano node doesnt crash. Wallet can't work with a database from previous version. 
Breaking changes again? :)

Should be nice, if there was some tool for DB mirgating between versions.

**Comment by stakeada** [comment]
However, I did rename the files in this directory (note the 2 in the name):
```
# ls ~/.local/share/cardano-wallet/jormungandr/testnet/chain/
blocks.sqlite-shm2  blocks.sqlite-wal2  blocks2.sqlite
```

Is there any other file that I need to rename or possibly even delete. From your message I only need to remove the old DB and resync and this problem should be gone?

**Comment by Fell-x27** [comment]
Yes, a new storage solves the problem.
I'm afraid that cause is not in filenames, but in database structure.

**Comment by stakeada** [comment]
I was saying that renaming the above didn't solve the issue.

So how can I solve the issue. Should I just remove the DBs, are these files the only ones that need to be removed:

- blocks.sqlite-shm
- blocks.sqlite-wal
- blocks.sqlite

Or do I need to do something else?

**Comment by rvl** [comment]
Hi, we are working on the wallet db schema migrations.

But for your information, the database structure is like this:

```
/home/rodney/.local/share/cardano-wallet/jormungandr/testnet
├── chain
│   ├── blocks.sqlite
│   ├── blocks.sqlite-shm
│   └── blocks.sqlite-wal
└── wallets
    ├── seq.071fc26f7f40d02cd2d2d3aee4ea900a14d4b266.sqlite
    ├── seq.071fc26f7f40d02cd2d2d3aee4ea900a14d4b266.sqlite-shm
    ├── seq.071fc26f7f40d02cd2d2d3aee4ea900a14d4b266.sqlite-wal
    ├── stake-pools.sqlite
    ├── stake-pools.sqlite-shm
    └── stake-pools.sqlite-wal

2 directories, 9 files
```

* The wallet databases (your data) are beneath the `wallets` directory.
   - Wallets are named with the wallet ID.
   - Stake pool metrics are in a separate file.
* The jormungandr node backend databases (chain data) are beneath the `chain` directory.
* The `-shm` and `-wal` files are [temporary files](https://www.sqlite.org/tempfiles.html) used by SQLite.

For simplicity, I recommend treating the lot as one unit. If you need to clear a database, clear it all and then restore your wallets from seed.

**Comment by KtorZ** [comment]
Hey, this one is actually a duplicate of #1181 and is not so much about db migration (though we are also working on it). More details on #1181 but 

TL;DR; Use master on a fresh new database

Cheers :)
